packages/pubsub/src/publisher.js

```diff
@@ -0,0 +1,146 @@
+/*!
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module pubsub/publisher
+ */
+
+'use strict';
+
+var arrify = require('arrify');
+var common = require('@google-cloud/common');
+var extend = require('extend');
+var is = require('is');
+var prop = require('propprop');
+
+/**
+ *
+ */
+function Publisher(topic, options) {
+  options = extend(true, {
+    batching: {
+      maxBytes: Math.pow(1024, 2) * 5,
+      maxMessages: 1000,
+      maxMilliseconds: 1000
+    }
+  }, options);
+
+  this.topic = topic;
+  this.api = topic.api;
+
+  this.inventory_ = {
+    queued: []
+  };
+
+  Object.defineProperty(this.inventory_, 'queueBytes', {
```

Comment:
This is a potential performance concern, as this needs to be checked every single time a message is added to the batch. This could be many, many times per second. I would recommend incrementing this value at message-addition time.